enricher/eks/metric.go

```diff
@@ -0,0 +1,91 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/mappings"
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/metricserver"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type EnricherMetric struct {
+	meter metric.Meter
+
+	outputRecordCount  metric.Int64Counter
+	outputDroppedCount metric.Int64Counter
+}
+
+func WithMetricServer(ms *metricserver.MetricServer) EnricherConfiguration {
+	return func(e *Enricher) error {
+		meter := ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks")
+		outputRecordCount, err := meter.Int64Counter("fluentbit.output.record", metric.WithDescription("output record counter"))
+
+		if err != nil {
+			return err
+		}
+
+		outputDroppedCount, err := meter.Int64Counter("fluentbit.output.dropped.body", metric.WithDescription("output dropped record counter where log or message does not exist"))
+
+		if err != nil {
+			return err
+		}
+
+		e.metric = &EnricherMetric{
+			meter:              ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks"),
+			outputRecordCount:  outputRecordCount,
+			outputDroppedCount: outputDroppedCount,
+		}
+
+		return nil
+	}
+}
+
+func (e *Enricher) AddRecordCount(record map[interface{}]interface{}, logType int) {
+	if e.metric == nil {
+		return
+	}
+
+	var serviceName = inferServiceName(record, logType)
+
+	e.metric.outputRecordCount.Add(context.TODO(), 1, metric.WithAttributes(attribute.Key(mappings.RESOURCE_SERVICE_NAME).String(serviceName)))
```

Comment:
Are we going to add a context in the future to `AddRecordCount`? if not we can change TODO to Background

Reply:
AFAIK, there is no intention but I could see the complexity of this growing (maybe the use of traces etc). But for now I will change this to background.